src/utils/tokens.ts

```diff
@@ -438,6 +487,34 @@ const tokens = {
       });
     }
 
+    if (depositAmount > 0) {
+      // get HTR deposit inputs
+      const selectedUtxos = await utxoSelection(storage, HATHOR_TOKEN_CONFIG.uid, depositAmount);
+      const foundAmount = selectedUtxos.amount;
+      for (const utxo of selectedUtxos.utxos) {
+        inputs.push(helpers.getDataInputFromUtxo(utxo));
+      }
+
+      if (foundAmount < depositAmount) {
+        throw new InsufficientFundsError(`Not enough HTR tokens for deposit: ${depositAmount} required, ${foundAmount} available`);
+      }
+
+      // get output change
+      if (foundAmount > depositAmount) {
+        const cAddress = await storage.getChangeAddress({ changeAddress });
+
+        outputs.push({
+          type: getAddressType(cAddress, storage.config.getNetwork()),
+          address: cAddress,
+          value: foundAmount - depositAmount,
+          timelock: null,
+          token: HATHOR_TOKEN_CONFIG.uid,
+          authorities: 0,
+          isChange: true,
+        });
+      }
+    }
```

Comment:
I'm guessing this part of the code is also present on other methods that involve depositing tokens...

Maybe open a KTLO to create an util method